File: src/Bunker/Bunker.Database/Entities/Task.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Bunker.Database.Entities
{
    public class Task
    {
        [Key]
        public int Id { get; set; }
        
        [Required]
        [StringLength(100)]
        public string Name { get; set; }
        
        [Required]
        [StringLength(2000)]
        public string Description { get; set; }
        
        public int Score { get; set; }
        
        [StringLength(200)]
        public string Answer { get; set; }
        
        public int ChallangeId { get; set; }
        
        [ForeignKey(nameof(ChallangeId))]
        public Challange Challange { get; set; }
    }
}

File: src/Bunker/Bunker.Domain/Task.fs
namespace Bunker.Domain

module Task =
    open Helpers
    open Domain

    let create player challangeId taskName taskDescription taskAnswer =
        match player.OwnedCompanies
              |> List.collect (fun x -> x.Challanges)
              |> List.tryFind (fun x -> x.Id = challangeId) with
        | Some challange -> 
            Ok { Id = NewTask
                 Name = taskName
                 Challange = challange
                 Description = taskDescription
                 Answer = taskAnswer }
        | None -> Fail [ "Challange not found" ]
